dask_image/ndmeasure/_utils/_label.py

```diff
@@ -202,44 +217,59 @@ def _chunk_faces(chunks, shape, structure):
     """
 
     ndim = len(shape)
-    numblocks = tuple(list(len(c) for c in chunks))
-    
+
     slices = da.core.slices_from_chunks(chunks)
-    
+
     # arrange block/chunk indices on grid
-    block_summary = np.arange(len(slices)).reshape(numblocks)
-    
-    faces = []
-    for ind_curr_block, curr_block in enumerate(np.ndindex(numblocks)):
-        
+    block_summary = np.arange(len(slices)).reshape(
+        [len(c) for c in chunks])
+
+    # Iterate over all blocks and use the structuring element
+    # to determine which blocks should be connected.
+    # For wrappped axes, we need to consider the block
+    # before the current block with index -1 as well.
+    numblocks = [len(c) if wrap_axes is None or ax not in wrap_axes
+                 else len(c) + 1 for ax, c in enumerate(chunks)]
+    for curr_block in np.ndindex(tuple(numblocks)):
+
+        curr_block = list(curr_block)
+
+        if wrap_axes is not None:
+            # start at -1 indices for wrapped axes
+            for wrap_axis in wrap_axes:
+                curr_block[wrap_axis] = curr_block[wrap_axis] - 1
+
+        # iterate over neighbors of the current block
         for pos_structure_coord in np.array(np.where(structure)).T:
-                        
+
             # only consider forward neighbors
             if min(pos_structure_coord) < 1 or \
                max(pos_structure_coord) < 2: continue
 
             neigh_block = [curr_block[dim] + pos_structure_coord[dim] - 1
                            for dim in range(ndim)]
 
-            if max([neigh_block[dim] >= numblocks[dim] for dim in range(ndim)]): continue
+            if max([neigh_block[dim] >= block_summary.shape[dim]
+                    for dim in range(ndim)]): continue
 
             # get neighbor slice index
-            ind_neigh_block = block_summary[tuple(neigh_block)]
+            ind_curr_block = block_summary[tuple(curr_block)]
```

Comment:
Added a small change to this comment in PR: https://github.com/dask/dask-image/pull/353